### PR TITLE
Remove usage of Custom error type

### PR DESCRIPTION
**Pull request by vihu**
- Introduce more specific errors when handling reports

### CODE REVIEW

File: file_store/src/error.rs [hunk 1]
```diff
@@ -24,6 +22,12 @@ pub enum Error {
     Config(#[from] config::ConfigError),
     #[error("mpsc channel error")]
     Channel,
+    #[error("unsupported datarate {0}")]
+    UnsupportedDataRate(String),
+    #[error("unsupported invalid_reason {0}")]
```
Comment (madninja) [inline file_store/src/error.rs:27]: may want to make this more explicit.. reason for what?

File: file_store/src/error.rs [hunk 2]
```diff
@@ -24,6 +22,12 @@ pub enum Error {
...
+    UnsupportedInvalidReason(String),
+    #[error("unsupported participant_side {0}")]
+    UnsupportedParticipantSide(String),
```
Comment (madninja) [inline file_store/src/error.rs:30]: May want to make this more explicit..

**Review by madninja — COMMENTED** [review]
I think this is good, but I'd suggest 1. pulling them into DecodeError since you're decoding the proto types into internal types and 2. making the names more explicit to about what they are.. at least in the string description for them

**Comment by vihu** [comment]
> I think this is good, but I'd suggest 1. pulling them into DecodeError since you're decoding the proto types into internal types and 2. making the names more explicit to about what they are.. at least in the string description for them

Check if [c688769](https://github.com/helium/oracles/pull/215/commits/c688769c8b664bac808b180d82c0e7adb3cdd09f) looks reasonable. Made it more verbose with specific type of proto and placed it in DecodeError.

**Review by madninja — COMMENTED** [review]
Better! I'd actually have `DecodeError::unsupported_datarate` as a function too which takes `Into<String>` for both the parameters to shorten the calling sites

**Comment by vihu** [comment]
> Better! I'd actually have `DecodeError::unsupported_datarate` as a function too which takes `Into<String>` for both the parameters to shorten the calling sites

Pushed [17f8f83](https://github.com/helium/oracles/pull/215/commits/17f8f835d46ab4e50d93f996417bb22b48acd534)

File: file_store/src/error.rs (hunk 1 above)
Comment (vihu) [reply file_store/src/error.rs]: [c688769](https://github.com/helium/oracles/pull/215/commits/c688769c8b664bac808b180d82c0e7adb3cdd09f)

File: file_store/src/error.rs (hunk 2 above)
Comment (vihu) [reply file_store/src/error.rs]: [c688769](https://github.com/helium/oracles/pull/215/commits/c688769c8b664bac808b180d82c0e7adb3cdd09f)

**Review by madninja — APPROVED** [review]

**Review by jeffgrunewald — APPROVED** [review]

**Pull request merged**